### PR TITLE
post CRUD operations initially implemented

### PR DESCRIPTION
**Pull request by mehmetemreakbulut**
***Description*:**
- Posts are entities that are fundamental for our forum page. It is very important for us to create them and send the data to the interface. In this direction, Get all posts, Get post and create post operations are implemented. 
- Importance is given to foreign keys here. I am trying to ensure that the data is sent to the interface correctly. This MR will remain open for the team to review. Tests will be added.

***Tasks Done*:**
- [x] An API with the name "get_all_posts" was created. It gets the authentication info and serves to the user all the posts.
- [x] An API with the name "get_post". It gets 'id' of the post as a path parameter and serves to the user the post related to this id.
- [x] An API with the name "create_post". It gets Post information containing 'title', 'body' and 'image_urls. Author information is derived from authentication credentials.
- [x] Pagination was added to the 'get_all_posts'. Pagination enables frontend and mobile developers to show the data partially and use smaller API responses.

***Reviewer*:**
@oguzhandemirelx 

**Notes:** 
- Development is still continuing. Tests will be added

***Issue*:** 
* See #290

### CODE REVIEW

File: app/backend/forum/models.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+class Post(models.Model):
+    title = models.CharField(max_length=100, null=False)
+    author = models.ForeignKey(CustomUser, null=False, on_delete=models.CASCADE)
+    body = models.TextField(null=False)
+    date = models.DateTimeField()
+    upvote = models.IntegerField(null=False, default=0)
+    downvote = models.IntegerField(null=False, default=0)
+
+class PostImages(models.Model):
+    image_url = models.CharField(max_length=100)
```
Comment (canberkboun9) [inline app/backend/forum/models.py:15]: I think 100 max length is not enough.

Comment (mehmetemreakbulut) [reply app/backend/forum/models.py:15]: > I think 100 max length may not be enough.

It is probably enough but to guarantee it, I will increase the maximum length constraint.

File: app/backend/forum/urls.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+from forum import views
+from rest_framework.authtoken.views import obtain_auth_token 
+from django.urls import path, include
+
+urlpatterns = [
+    path('posts', views.get_all_posts, name='get_all_posts'),
+    path('post', views.create_post, name='create post'),
```
Comment (canberkboun9) [inline app/backend/forum/urls.py:7]: Is naming convention here intentional?

Comment (mehmetemreakbulut) [reply app/backend/forum/urls.py:7]: > Is naming convention here intentional?

It is a mistake. Good feedback. Thanks

File: app/backend/forum/views.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+def create_post(request):
+    title = request.data['title']
+    author = request.user
+    body = request.data['body']
+    date = datetime.now()
+
+    post = Post(title=title, author=author, body=body, date=date)
+    post.save()
+
+    image_urls = request.data['image_urls']
```
Comment (canberkboun9) [inline app/backend/forum/views.py:46]: We deal with the images in POST but not in GET requests here. I wonder if it will cause a problem.

**Review by canberkboun9 — COMMENTED** [review]
- I added my comments on the commit as line comments.
- Emre responded them in a short time, even before I finished the review.
- I have no other feedback on this commit for now.
- Oğuzhan is welcome to make a review as well.
- I am not merging it.

**Comment by oguzhandemirelx** [comment]
For now, these endpoints are sufficient. Your commit is excellent; I have nothing to add to Canberk's remarks. I appreciate your work.

**Pull request merged**